crates/rust-analyzer/src/handlers.rs

```diff
@@ -621,14 +621,17 @@ pub(crate) fn handle_prepare_rename(
     let _p = profile::span("handle_prepare_rename");
     let position = from_proto::file_position(&snap, params)?;
 
-    let optional_change = snap.analysis.rename(position, "dummy")?;
-    let range = match optional_change {
-        None => return Ok(None),
-        Some(it) => it.range,
+    let change = match snap.analysis.rename(position, "dummy")? {
+        Ok(optional_change) => optional_change,
+        Err(err) => {
+            return Ok(Some(PrepareRenameResponse::RangeWithPlaceholder {
+                range: Range::default(),
+                placeholder: err.to_string(),
+            }))
```

Comment:
I think we should return an `Err` response here, according to the protocol?

https://microsoft.github.io/language-server-protocol/specification#textDocument_prepareRename

> error: code and message set in case the element can’t be renamed. Clients should show the information in their user interface.